local/compose/build.go

```diff
@@ -161,7 +176,37 @@ func (s *composeService) toBuildOptions(service types.ServiceConfig, contextPath
 		Tags:      tags,
 		Target:    service.Build.Target,
 		Exports:   []bclient.ExportEntry{{Type: "image", Attrs: map[string]string{}}},
+		Platforms: platforms,
+	}, nil
+}
+
+func getPlatforms(ps ...string) ([]specs.Platform, error) {
+	var platforms []specs.Platform
+	for _, p := range ps {
+		platform, err := getPlatform(p)
+		if err != nil {
+			return nil, err
+		}
+		platforms = append(platforms, platform)
+	}
+	return platforms, nil
+}
+
+func getPlatform(p string) (specs.Platform, error) {
+	if p == "" {
+		return specs.Platform{
+			OS:           runtime.GOOS,
+			Architecture: runtime.GOARCH,
+		}, nil
+	}
+	tokens := strings.Split(p, "/")
+	if len(tokens) != 2 {
+		return specs.Platform{}, errdefs.InvalidParameter(errors.New("could not parse platform " + p))
 	}
+	return specs.Platform{
+		OS:           tokens[0],
+		Architecture: tokens[1],
+	}, nil
```

Comment:
for consistency, this should be implemented as a `Transform` func by compose-go, like other "short syntax" strings vs dict.

Reply:
I could actually remove this, since it's already implemented in `containerd`.
PTAL